### PR TITLE
handle silent notifications to generate local notifications with message plaintext CORE-5775

### PR DESCRIPTION
**Pull request by mmaxim**
Patch does the following:

1.) Create a new local endpoint `UnboxMobilePushNotification`, whose job is to take `chat1.MessageBoxed` from the push server and unbox and stash in the body cache.
2.) Give `UnboxMobilePushNotification` the power to cancel audible notifications if we successfully process the background notification. We do this by just checking to see if we made it through the process with no error, and then hitting a special endpoint on Gregor. We do not use the existing Gregor connection handler, since we don't want to run the whole sync procedure (to conserve battery and not get blacklisted by iOS), and instead just make a new connection that we immediatley destroy.
3.) Add functionality to the frontend to process the new background notifications, and to call `UnboxMobilePushNotification`. Upon success of this function, then show a local notification with the plaintext of the message.

### CODE REVIEW

**Comment by mmaxim** [comment]
cc @malgorithms

**Comment by keybase-ci-visdiff** [comment]
The commits c160e4d84912...0c110fdfd95f introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1501878855526-c160e4d84912-0c110fdfd95f-linux/Markdown-emoji-c160e4d84912.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1501878855526-c160e4d84912-0c110fdfd95f-linux/Markdown-emoji-0c110fdfd95f.png)  
   Markdown-emoji comparison
</summary>

File: protocol/avdl/chat1/local.avdl [hunk 1]
```diff
@@ -661,19 +661,22 @@ protocol local {
...
   record SetAppNotificationSettingsLocalRes {
     boolean offline;
-    array<RateLimit> rateLimits; 
-  } 
-  SetAppNotificationSettingsLocalRes setAppNotificationSettingsLocal(ConversationID convID, ConversationNotificationInfo settings); 
+    array<RateLimit> rateLimits;
+  }
+  SetAppNotificationSettingsLocalRes setAppNotificationSettingsLocal(ConversationID convID, ConversationNotificationInfo settings);
+
+  // Unpack message form a push notification
```
Comment (patrickxb) [inline protocol/avdl/chat1/local.avdl]: from

**Comment by keybase-ci-visdiff** [comment]
The commits 18253ef34523...2784f9c0bc15 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1501884765421-18253ef34523-2784f9c0bc15-linux/Markdown-emoji-18253ef34523.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1501884765421-18253ef34523-2784f9c0bc15-linux/Markdown-emoji-2784f9c0bc15.png)  
   Markdown-emoji comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 18253ef34523...31e370337cf4 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1501886524996-18253ef34523-31e370337cf4-linux/Markdown-emoji-18253ef34523.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1501886524996-18253ef34523-31e370337cf4-linux/Markdown-emoji-31e370337cf4.png)  
   Markdown-emoji comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 18253ef34523...15ac3c045400 introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Profile-Your_Profile_Broken** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502067569981-18253ef34523-15ac3c045400-linux/Profile-Your_Profile_Broken-18253ef34523.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502067569981-18253ef34523-15ac3c045400-linux/Profile-Your_Profile_Broken-15ac3c045400.png)  
   Profile-Your_Profile_Broken comparison
 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502067569981-18253ef34523-15ac3c045400-linux/Markdown-emoji-18253ef34523.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502067569981-18253ef34523-15ac3c045400-linux/Markdown-emoji-15ac3c045400.png)  
   Markdown-emoji comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 1130bbb47b56...96296b83f71b introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502067724056-1130bbb47b56-96296b83f71b-linux/Markdown-emoji-1130bbb47b56.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502067724056-1130bbb47b56-96296b83f71b-linux/Markdown-emoji-96296b83f71b.png)  
   Markdown-emoji comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 1130bbb47b56...accbd9d6cc75 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502070980989-1130bbb47b56-accbd9d6cc75-linux/Markdown-emoji-1130bbb47b56.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1502070980989-1130bbb47b56-accbd9d6cc75-linux/Markdown-emoji-accbd9d6cc75.png)  
   Markdown-emoji comparison
</summary>

File: shared/constants/push.js [hunk 2]
```diff
@@ -19,6 +19,11 @@ export type PushNotification = {
       convID?: string,
       type?: string,
       username?: string,
+      c?: string,
```
Comment (chrisnojima) [inline shared/constants/push.js]: http://db3.memegenerator.net/cache/instances/folder238/500x/80120238.jpg

Comment (mmaxim) [inline shared/constants/push.js]: Hah, it's part of the payload, which is limited to 2048 bytes, so I wanted to make it as small as possible.

**Comment by mmaxim** [comment]
Thanks! Will wait for the merge.

**Pull request merged**